### ISSUE TITLE
Add documentation

### ISSUE THREAD

**Issue by pszuster** [jam01/java-mule-agent#2]
It would be useful to have some docs on how to use this agent

**Comment by jam01** [comment]
Yeah, docs and examples are in the horizon. Opentracing's Java Agent was deprecated however, the new project Special Agent is starting to achieve some maturity so hopefully soon I'll have some time to migrate to that and then work on a better demo.

**Comment by pszuster** [comment]
Any quick instructions on how to use ir as-is with mule 4?

**Comment by jam01** [comment]
Ah more or less you have to build a JAR from this project, and you want to start the mule instance with a reference to the JAR like this:

```
wrapper.java.additional.17=-javaagent:"%OT_HOME%/opentracing-mule-agent.jar"
wrapper.java.additional.17.stripquotes=TRUE
```
Also, you'll need to provide environment variables for the Jaeger tracer to find the collector, you can check out the jaeger docs for the possible variables but the demo I had used

```
JAEGER_SAMPLER_MANAGER_HOST_PORT
JAEGER_AGENT_HOST
JAEGER_SERVICE_NAME
```

Currently this project packages Jaeger because that was the easiest way to get a POC working, but eventually the tracer implementation should definitely be provided by the user in some way or another.
